examples/tts/magpietts_inference.py

```diff
@@ -0,0 +1,603 @@
+    eval_group = parser.add_argument_group('Evaluation')
+    eval_group.add_argument(
+        '--run_evaluation',
+        action='store_true',
+        help='Run evaluation after inference (default: False, inference only)',
+    )
+    eval_group.add_argument('--sv_model', type=str, default="titanet", choices=["titanet", "wavlm"])
+    eval_group.add_argument('--asr_model_name', type=str, default="nvidia/parakeet-tdt-1.1b")
+    eval_group.add_argument('--num_repeats', type=int, default=1)
+    eval_group.add_argument('--confidence_level', type=float, default=0.95)
+    eval_group.add_argument('--disable_utmosv2', action='store_true')
+    eval_group.add_argument(
+        '--violin_plot_metrics',
+        type=str,
+        nargs='*',
+        default=['cer', 'pred_context_ssim', 'utmosv2'],
+    )
+
+    # Quality targets (for CI/CD)
+    target_group = parser.add_argument_group('Quality Targets')
+    target_group.add_argument('--cer_target', type=float, default=None)
+    target_group.add_argument('--ssim_target', type=float, default=None)
+
+    return parser
+
+
+def main():
+    """Main entry point."""
+    parser = create_argument_parser()
+    args = parser.parse_args()
+
+    # Set default datasets if not provided
+    if args.datasets is None:
+        args.datasets = EVALUATION_DATASETS
+
+    datasets = args.datasets.split(",")
+
+    # Determine mode and validate
+    has_checkpoint_mode = (
+        args.hparams_files is not None
+        and args.checkpoint_files is not None
+        and args.hparams_files != "null"
+        and args.checkpoint_files != "null"
+    )
+    has_nemo_mode = args.nemo_files is not None and args.nemo_files != "null"
+
+    if not has_checkpoint_mode and not has_nemo_mode:
+        parser.error("You must provide either:\n" "  1. --hparams_files and --checkpoint_files\n" "  2. --nemo_files")
+
+    # Build configurations
+    inference_config = InferenceConfig(
+        temperature=args.temperature,
+        topk=args.topk,
+        batch_size=args.batch_size,
+        use_cfg=args.use_cfg,
+        cfg_scale=args.cfg_scale,
+        apply_attention_prior=args.apply_attention_prior,
+        attention_prior_epsilon=args.attention_prior_epsilon,
+        attention_prior_lookahead_window=args.attention_prior_lookahead_window,
+        estimate_alignment_from_layers=parse_layer_list(args.estimate_alignment_from_layers),
+        apply_prior_to_layers=parse_layer_list(args.apply_prior_to_layers),
+        start_prior_after_n_audio_steps=args.start_prior_after_n_audio_steps,
+        use_local_transformer=args.use_local_transformer,
+        maskgit_n_steps=args.maskgit_n_steps,
+        maskgit_noise_scale=args.maskgit_noise_scale,
+        maskgit_fixed_schedule=args.maskgit_fixed_schedule,
+        maskgit_sampling_type=args.maskgit_sampling_type,
+        eos_detection_method=args.eos_detection_method,
+        ignore_finished_sentence_tracking=args.ignore_finished_sentence_tracking,
+    )
```

Comment:
We can simplify the `dataclass` initialization by defining a new class function as `from_dict(cls, config)` and do `inference_config = InferenceConfig.from_dict(args)`

**Demo Code**
```python
from dataclasses import dataclass, fields

@dataclass
class InferenceConfig:
    ...

    @classmethod
    def from_dict(cls, config):
        field_names = {f.name for f in fields(cls)}
        config_new = {}
        for k, v in config.items():
            if k in field_names:
                if k in ["estimate_alignment_from_layers", "apply_prior_to_layers"]:
                    v = parse_layer_list(v) 
                config_new[k] = v
        return cls(**config_new)

inference_config = InferenceConfig.from_dict(vars(args))
```